geode-core/src/main/java/org/apache/geode/cache/query/internal/CompiledComparison.java

```diff
@@ -296,7 +296,7 @@ public boolean isRangeEvaluatable() {
   public int getSizeEstimate(ExecutionContext context) throws FunctionDomainException,
       TypeMismatchException, NameResolutionException, QueryInvocationTargetException {
     IndexInfo[] idxInfo = getIndexInfo(context);
-    if (idxInfo == null) {
+    if ((idxInfo == null) || (idxInfo.length > 1)) {
```

Comment:
Just a question, should this condition be on it's own and does it make sense to give it a score of 0?  Should these types of conditions be waited more heavily or considered "better?"  I'm ok with 0 just curious if you tried different cases out :-)

Reply:
Good point, and I think you're right... having more than one index available for a `CompiledComparison` implies that both the left and right operands are indexed, so that case should have more "weight" when compared to an independent operator that would imply scanning the full region.
Will make the changes and add some more tests, thanks for finding this!.

Reply:
Done!